virtual-classroom/src/main/java/ro/utcn/sd/mid/assign1/virtualclassroom/controller/ConsoleController.java

```diff
@@ -0,0 +1,360 @@
+        String editDeleteWord = value ? "edit" : "delete";
+        print("Id of the answer to " + editDeleteWord + ": ");
+        String id = scanner.next();
+        try {
+            int aId = Integer.parseInt(id);
+            try {
+                Answer answer = answerService.findById(aId);
+                if (answer.getUserId().equals(loggedInUser.getId())) {  // verify the creator
+                    if (value) {    // choose between edit and delete, true means edit
+                        print("New text of the answer: ");
+                        String newText = scanner.next();
+                        answer.setText(newText);
+                        answerService.saveAnswer(answer);
+                    } else {
+                        answerService.deleteAnswer(answer);
+                    }
+                    println("Your answer was " + editDeleteWord + "ed.");
+                } else {
+                    println("You are not the creator of this answer. You cannot " + editDeleteWord + " it.");
+                }
+            } catch (AnswerNotFoundException e) {
+                println("There is no question with id = " + aId);
+            }
+        } catch (NumberFormatException e) {
+            println("Invalid id. Please introduce a number.");
+        }
+    }
+
+
+    // ================== Feature 3 handlers ==================
+
+    private void handleQuestionVote(boolean value) {
+        String voteType = value ? "upvote" : "downvote";
+        print("Id of the question to " + voteType + ": ");
+        String id = scanner.next();
+        try {
+            int qId = Integer.parseInt(id);
+            try {
+                Question toBeVoted = questionService.findById(qId);
+                if (toBeVoted.getId().equals(loggedInUser.getId())) {
+                    println("You cannot " + voteType + " your own question.");
+                } else {
+                    try {
+                        voteService.voteQuestion(qId, loggedInUser.getId(), value);
+                        println("Your " + voteType + " was registered.");
+                    } catch (AlreadyVotedException e) {
+                        println("You cannot " + voteType + " multiple times the same question.");
+                    }
+
+                }
+            } catch (QuestionNotFoundException e) {
+                println("There is no question with id = " + qId);
+            }
+        } catch (NumberFormatException e) {
+            println("Invalid id. Please introduce a number.");
+        }
+    }
+
+    private void handleAnswerVote(boolean value) {
+        String voteType = value ? "upvote" : "downvote";
+        print("Id of the answer to " + voteType + ": ");
+        String id = scanner.next();
+        try {
+            int aId = Integer.parseInt(id);
+            try {
+                Answer toBeVoted = answerService.findById(aId);
+                if (toBeVoted.getId().equals(loggedInUser.getId())) {
```

Comment:
Also, these kinds of validations are business rules. They should reside inside the service layer.